### PR TITLE
Updates for compat with SDK 1.2.0

### PR DESCRIPTION
**Pull request by jmrossy**

### CODE REVIEW

File: contracts/HypERC20Collateral.sol [hunk 1]
```diff
@@ -15,9 +16,8 @@ contract HypERC20Collateral is TokenRouter {
     /**
      * @notice Constructor
      * @param erc20 Address of the token to keep as collateral
-     * @param gasAmount Amount of destination gas to be paid for processing
      */
-    constructor(address erc20, uint256 gasAmount) TokenRouter(gasAmount) {
+    constructor(address erc20) {
```
Comment (nambrot) [inline contracts/HypERC20Collateral.sol:20]: Where is the gasAmount now set?

File: src/app.ts [hunk 2]
```diff
@@ -1,43 +1,64 @@
...
+> extends GasRouterApp<Contracts> {
+  async transfer(
+    origin: ChainName,
+    destination: ChainName,
+    recipient: Address,
+    amountOrId: BigNumberish,
+  ) {
+    const originRouter = this.getContracts(origin).router;
+    const destinationDomain = this.multiProvider.getDomainId(destination);
+    const gasPayment = await originRouter.quoteGasPayment(destinationDomain);
```
Comment (nambrot) [inline src/app.ts:20]: This technically won't work on the audited version of warp routes. Dont have to fix it here, but just flagging

File: src/app.ts [hunk 3]
```diff
@@ -1,43 +1,64 @@
...
+export class HypERC20App extends HyperlaneTokenApp<HypERC20Contracts> {
+  async transfer(
+    origin: ChainName,
+    destination: ChainName,
+    recipient: Address,
+    amount: BigNumberish,
+  ) {
+    const originRouter = this.getContracts(origin).router;
+    const signerAddress = await this.multiProvider.getSignerAddress(origin);
+    const balance = await originRouter.balanceOf(signerAddress);
```
Comment (nambrot) [inline src/app.ts:39]: Ditto

**Review by nambrot — APPROVED** [review]
Look a bit weird with some of the gas router changes sprinkled in here, so I would probably get @yorhodes 's approval

**Review by yorhodes — APPROVED** [review]

**Pull request merged**